R/measures.R

```diff
@@ -1415,27 +1415,27 @@ iauc.uno = makeMeasure(id = "iauc.uno", minimize = FALSE, best = 1, worst = 0,
 #' @rdname measures
 #' @format none
 ibrier = makeMeasure(id = "ibrier", minimize = TRUE, best = 0, worst = 1,
-  properties = c("surv", "req.truth", "req.model", "req.task"),
+  properties = c("surv", "req.prob", "req.feats"),
   name = "Integrated brier score using Kaplan-Meier estimator for weighting",
   note = "Only works for methods for which probabilities are provided via pec::predictSurvProb. Currently these are only coxph and randomForestSRC. To set an upper time limit, set argument max.time (defaults to max time in test data). Implemented in pec::pec",
   fun = function(task, model, pred, feats, extra.args) {
     requirePackages(c("survival", "pec"))
-    targets = getTaskTargets(task)
-    tn = getTaskTargetNames(task)
-    f = as.formula(sprintf("Surv(%s, %s) ~ 1", tn[1L], tn[2L]))
-    newdata = getTaskData(task)[model$subset, ]
-    max.time = extra.args$max.time %??% max(newdata[[tn[1L]]])
-    grid = seq(0, max.time, length.out = extra.args$resolution)
-
-    probs = predictSurvProb(getLearnerModel(model, more.unwrap = TRUE), newdata = newdata, times = grid)
-    perror = pec(probs, f, data = newdata[, tn], times = grid, exact = FALSE, exactness = 99L,
-      maxtime = max.time, verbose = FALSE, reference = FALSE)
-    
-
-    # FIXME: this might be the wrong number!
-    crps(perror, times = max.time)[1L, ]
-  },
-  extra.args = list(max.time = NULL, resolution = 1000L)
+    truth = getPredictionTruth(pred)
+    target = getPredictionTaskDesc(pred)$target
+    newdata = cbind(truth[,1], pred$data$truth.event, feats)
+    colnames(newdata)[1:2] = target
+    probs = getPredictionProbabilities(pred)
+    if (anyMissing(probs))
+      return(NA_real_)
+    colnames(probs) = times.train = as.numeric(substr(colnames(probs), 11, 100))
```

Comment:
Same as above: Is there a need to save the training times at this point? Why not use `time.train` as a normal measure?

E: Ah ok, its about survial times. I have no knowledge in this.

Reply:
This is necessary, to get the information about the training times to the model (we predict for each training and prediction time that exist).